sherman adagram

Peacegypsy opened the pull request:
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | iterating through arrays/hashes, case/switch statements.
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | different specificity, similar tactics.
What was a challenge you were able to overcome on this assignment? | finding new answers to challenges i had in ruby with this project, like working with hashes and arrays and iterating through them. 
What is your favorite thing about learning a new programming language? | When something runs correctly or when you understand what to write to make it work.
What is your least favorite thing about learning a new programming language? | breaking habits or patterns built in Ruby and replacing them with new ones.
Do you have any recommendations on how we could improve this project for the next cohort? | let us know a few more functions before we get started, or a recommended resource for looking them up.

Commit:
checking to see if the code runs, yes, all tests failing

Commit:
figuring out what works in removing letters from pool

Commit:
fixed usesavailableletters

Commit:
working on scoring words

Commit:
6 tests passing. frustrated.

Commit:
fixing tests after tutoring help

CheezItMan commented on src/adagrams.js:
```diff
@@ -1,8 +1,241 @@
+      .toUpperCase()
+      .split("")
+      .reduce((wordScore, letter) => {
+        const letterScore = Adagrams.scoreCHART[letter];
```
I believe this should be:  `        const letterScore = Adagrams.SCORE_CHART[letter];`

Otherwise good use of `reduce` and `split`.

CheezItMan commented on src/adagrams.js:
```diff
@@ -1,8 +1,241 @@
+    let tilesDrawn = 0;
+
+    while (tilesDrawn < 10) {
+      let rand = letterPool[Math.floor(Math.random() * letterPool.length)];
```
I'd prefer `const`

CheezItMan commented on src/adagrams.js:
```diff
@@ -1,8 +1,241 @@
+    const availableLetters = input.split("");
+    const handCopy = hand.slice(0);
+
+    return availableLetters.every(letter => {
```
Nice use of `.every`

CheezItMan commented:
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions |  Check, for JS functions, checking out the MDN is a good start, or using Google to search for solutions.
Small commits with meaningful commit messages |  Good commit messages and good number of commits
**Code Requirements** |
`drawLetters` method |  Check
Uses appropriate data structure to store the letter distribution |  Check
All tests for `drawLetters` pass |  Check
`usesAvailableLetters` method |  Check
All tests for `usesAvailableLetters` pass |  Check
`scoreWord` method | Check, with a typo, see my notes.  Otherwise it's really nicely done.
Uses appropriate data structure to store the letter scores |  Check
All tests for `scoreWord` pass |  With the typo fixed, yes.
`highestScoreFrom` method |  **MISSING**
Appropriately handles edge cases for tie-breaking logic |  **MISSING**
All tests for `highestScoreFrom` pass |  **MISSING**
**Overall** |  You hit **most** of the learning goals for the project.  The `scoreWord` method is very well designed.  Lets sit down this week and review the project, and see how the remaining methods could be done.

Commit:
added highest word score function

Peacegypsy commented:
Hi Chris,
I added my highestScore function. Can you check that its okay please?

-And thank you for pointing out the typo! I’ve been trying to translate my original version in javascript and I get my ’names’ mixed up.

Christina

> On Nov 25, 2018, at 11:04 PM, Chris M <notifications@github.com> wrote:
> 
> @CheezItMan commented on this pull request.
> 
> In src/adagrams.js <https://github.com/Ada-C10/js-adagrams/pull/46#discussion_r236138101>:
> 
> > +    W: 4,
> +    Y: 4,
> +    K: 5,
> +    J: 8,
> +    X: 8,
> +    Q: 10,
> +    Z: 10
> +  },
> +
> +  LENGTH_BONUS: 8,
> +  scoreWord(word) {
> +    return word
> +      .toUpperCase()
> +      .split("")
> +      .reduce((wordScore, letter) => {
> +        const letterScore = Adagrams.scoreCHART[letter];
> I believe this should be: const letterScore = Adagrams.SCORE_CHART[letter];
> 
> Otherwise good use of reduce and split.
> 
> In src/adagrams.js <https://github.com/Ada-C10/js-adagrams/pull/46#discussion_r236138398>:
> 
> > +    //     W: 4,
> +    //     Y: 4,
> +    //     K: 5,
> +    //     J: 8,
> +    //     X: 8,
> +    //     Q: 10,
> +    //     Z: 10,
> +    //
> +    //     LENGTH_BONUS: 8
> +    //   };
> +
> +    const hand = [];
> +    let tilesDrawn = 0;
> +
> +    while (tilesDrawn < 10) {
> +      let rand = letterPool[Math.floor(Math.random() * letterPool.length)];
> I'd prefer const
> 
> In src/adagrams.js <https://github.com/Ada-C10/js-adagrams/pull/46#discussion_r236138646>:
> 
> > +  //     const drawnCopy = drawnLetters.slice(0); // Create a copy of the drawn letters which we can modify
> +  //
> +  //     return letters.every(letter => {
> +  //       const index = drawnCopy.indexOf(letter);
> +  //       if (index === -1) return false;
> +  //
> +  //       delete drawnCopy[index];
> +  //       return true;
> +  //     });
> +  //   }
> +  // };
> +  usesAvailableLetters(input, hand) {
> +    const availableLetters = input.split("");
> +    const handCopy = hand.slice(0);
> +
> +    return availableLetters.every(letter => {
> Nice use of .every
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/Ada-C10/js-adagrams/pull/46#pullrequestreview-178150378>, or mute the thread <https://github.com/notifications/unsubscribe-auth/Ab0PkpPxyMK-AV-qusuPWskv-C3tiiuiks5uy5J9gaJpZM4YyaCp>.
>
